### PR TITLE
Dev/6 ramuh

### PR DESCRIPTION
**Pull request by clpetersonucf**
** THIS INCLUDES MIGRATIONS **

- Sanitizes widget title when exporting play data (#1235)
- Rewords "spoof" to "mock" in tests (#1141)
- Collaboration access level now set to a default value instead of blank (#1158)
- Fixes to long title styling in LTI widget selection (#1129)
- Modest additions to admin page (#1146)
- Relocates logged in user to right side of header (#1214)
- Adds export option for referrer URLs (#176)
-  Guide page will only display tabs for existing guides (player/creator) (#1249)
- Adds clarifying information to Export Options modal concerning LTI launches (#1216)
- Support for participation score logging (#1217) (REQUIRES MIGRATION)
- Removes Group field for widgets (#707) (REQUIRES MIGRATION)

### CODE REVIEW

**Comment by clpetersonucf** [comment]
Ready for review.

**Comment by clpetersonucf** [comment]
✅ default collaboration level works
✅ relocated login button works and looks good
✅ referrer url export option works
✅ LTI scores clarifying info on High Scores export works and looks good

Still testing the rest

**Comment by FrenjaminBanklin** [comment]
Checked/confirmed:
- [x] Sanitizes widget title when exporting play data (#1235)
- [x] Rewords "spoof" to "mock" in tests (#1141)
- [x] Collaboration access level now set to a default value instead of blank (#1158)
- [x] Fixes to long title styling in LTI widget selection (#1129) SEE BELOW
- [x] Modest additions to admin page (#1146)
- [x] Relocates logged in user to right side of header (#1214)
- [x] Adds export option for referrer URLs (#176)
- [x] Guide page will only display tabs for existing guides (player/creator) (#1249)
- [x] Adds clarifying information to Export Options modal concerning LTI launches (#1216)
Still need to check:
- [ ] Support for participation score logging (#1217) (REQUIRES MIGRATION)
- [ ] Removes Group field for widgets (#707) (REQUIRES MIGRATION)

---
Re: long titles in the LTI picker, really long widget titles are still just a little problematic, see
Screen Shot 2019-08-06 at 11 37 43 AM
and
Screen Shot 2019-08-06 at 11 37 56 AM

This doesn't prevent anything from working, it just looks a little gross - and I don't see many people doing this, even accidentally.

**Comment by clpetersonucf** [comment]
Looking to get approvals for this. Ran through all the additions on QA and everything appears to be in order.

File: fuel/app/classes/materia/session/playdataexporter.php [hunk 1]
```diff
@@ -376,4 +376,62 @@ protected static function questions_and_answers($inst, $semesters)
 
 		return [$string, '_questions_answers.csv'];
 	}
+
+	// Outputs a .zip file of two CSV files for individual and collective referrers data
+	protected static function referrer_links($inst, $semesters)
```
Comment (iturgeon) [inline fuel/app/classes/materia/session/playdataexporter.php]: Change the export dropdown to read `Referrer Urls`

File: fuel/app/classes/materia/session/log.php [hunk 2]
```diff
@@ -36,6 +36,7 @@ class Session_Log
 	const TYPE_SCORE_ALERT             = 'SCORE_ALERT';
 	const TYPE_SCORE_FEEDBACK          = 'SCORE_FEEDBACK';
 	const TYPE_SCORE_SET_QUESTION      = 'SCORE_SET_QUESTION';
+	const TYPE_UNSCORABLE              = 'SCORE_UNSCORABLE';
```
Comment (iturgeon) [inline fuel/app/classes/materia/session/log.php]: Before this makes it in, I want to ask the group if `SCORE_UNSCORABLE` and `TYPE_UNSCORABLE` are the best labels for this?

I'm thinking  `TYPE_PARTICIPATION_SCORE` and `PARTICIPATION_SCORE` seems more accurate?

File: fuel/app/tests/api/v1.php [hunk 3]
```diff
@@ -625,6 +625,68 @@ public function test_widget_publish_perms_verify(): void
 		$this->assertTrue($output);
 	}
 
+	public function test_participation_score_equals_input_value()
+	{
+		$this->markTestIncomplete(); // gotta make sure it ws made
+		return;
```
Comment (iturgeon) [inline fuel/app/tests/api/v1.php]: This test is short circuited? intentional?

Comment (weining-li) [reply fuel/app/tests/api/v1.php]: Mmm... a little bit intentional? The rest of the test passed but I saw that the `test_session_play_create` method had these lines so I also put them here afterwards.

File: fuel/app/classes/materia/session/log.php (hunk 2 above)
Comment (clpetersonucf) [reply fuel/app/classes/materia/session/log.php]: `TYPE_UNSCORABLE` is now `TYPE_SCORE_PARTICIPATION`
`SCORE_UNSCORABLE` is now `SCORE_PARTICIPATION`

File: fuel/app/tests/api/v1.php (hunk 3 above)
Comment (clpetersonucf) [reply fuel/app/tests/api/v1.php]: I removed the short circuit and made sure the test passes, presumably it's working as intended now.

File: fuel/app/classes/materia/session/playdataexporter.php (hunk 1 above)
Comment (clpetersonucf) [reply fuel/app/classes/materia/session/playdataexporter.php]: Done. Now appears as `Referrer URLs`

File: fuel/app/themes/default/partials/header.php [hunk 4]
```diff
@@ -17,10 +17,20 @@
...
 	<h1 class="logo"><a href="/">Materia</a></h1>
 
-	<span ng-if="currentUser.loggedIn">
+	<span ng-switch="currentUser.loggedIn">
 		<p class="user avatar">
-			<img ng-src="{{currentUser.avatar}}" />
-			Welcome <a href="/profile">{{currentUser.name}}</a>
+			<a ng-switch-when="true" href="/profile">
+				<span>{{currentUser.name}}</span>
+				<img ng-src="{{currentUser.avatar}}" />
```
[the remark this answers is not in the document]
Comment (clpetersonucf) [reply fuel/app/themes/default/partials/header.php]: Modest tweaks: Screen Shot 2019-08-22 at 2 55 41 PM

**Pull request merged**